R/system.R

```diff
@@ -2993,6 +3007,35 @@ searchAndReadExcelFiles <- function(folder, forPreview = FALSE, pattern = "", sh
 
 }
 
+#'API that imports multiple same structure Excel sheets in the Excel Book and merge them to a single data frame
+#'@export
+read_excel_file_multi_sheets <- function(file, forPrevew = FALSE, sheets = c(1), col_names = TRUE, col_types = NULL, na = "", skip = 0, trim_ws = TRUE, n_max = Inf, use_readxl = NULL, detectDates = FALSE, skipEmptyRows = FALSE, skipEmptyCols = FALSE, check.names = FALSE, tzone = NULL, convertDataTypeToChar = TRUE, ...) {
+  # set name to the files so that it can be used for the "id" column created by purrr::map_dfr.
+  sheets <- setNames(as.list(sheets), sheets)
+  df <- purrr::map_dfr(sheets, .f = ~exploratory::read_excel_file(
+                       path = file,
+                       sheet = .x,
+                       col_names = col_names,
+                       col_types = col_types,
+                       na = na,
+                       skip = skip,
+                       trim_ws = trim_ws,
+                       n_max = n_max,
+                       use_readxl = use_readxl,
+                       detectDates = detecDates,
+                       skipEmptyRows = skipEmptyRows,
+                       skipEmptyCols = skipEmptyCols,
+                       check.names = check.names,
+                       tzone = tzone, convertDataTypeToChar = convertDataTypeToChar),
+                       .id = "exp.file.id") %>% mutate(exp.file.id = basename(exp.file.id)) # extract file name from full path with basename.
```

Comment:
Rename the column name exp.file.id appropriately.
basename() is not necessary.

Reply:
I renamed the "exp.file.id" to "exp.sheet.id". 
I also removed the basename() mutate command.